sklearn/manifold/t_sne.py

```diff
@@ -714,6 +714,11 @@ def _fit(self, X, skip_num_points=0):
             if self.verbose:
                 print("[t-SNE] Computing {} nearest neighbors...".format(k))
 
+            if self.metric == "precomputed" and sp.issparse(X):
+                if np.any(getnnz(X, axis=1) < k):
+                    raise ValueError("Perplexity of the sparse matrix is high."
```

Comment:
The perplexity is not of the sparse matrix. what you mean is something like: "{} neighbors per sample are required for the current perplexity. Decrease perplexity, or provide more precomputed distances per sample."